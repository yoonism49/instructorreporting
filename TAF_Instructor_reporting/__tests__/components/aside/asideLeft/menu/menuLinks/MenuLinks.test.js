'use strict';

import React            from 'react';
import MenuLinks        from '../../../../../../src/app/components/aside/asideLeft/menu/menuLinks/MenuLinks';
import renderer         from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';

let activeViewMock;
let onClick;

describe('AnimatedView component', () => {
  const mockViews = [
    { name: 'view1', linkTo: '/view1', faIconName: 'fa-home', itemCount: 1 },
    { name: 'view2', linkTo: '/view2', faIconName: 'fa-user', itemCount: 0 }
  ];

  beforeEach(() => {
    activeViewMock = mockViews[0].name;
    onClick = jest.fn();
  });

  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <MemoryRouter>
          <MenuLinks
            activeView={activeViewMock}
            views={mockViews}
            onClick={onClick}
          />
        </MemoryRouter>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
