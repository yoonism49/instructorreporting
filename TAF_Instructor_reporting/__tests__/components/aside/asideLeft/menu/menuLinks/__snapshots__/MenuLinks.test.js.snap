// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedView component renders as expected 1`] = `
<div>
  <ul
    className="sidebar-menu sidebar-menu__marginTop"
  >
    <li
      className="active"
    >
      <a
        href="/view1"
        onClick={[Function]}
      >
        <i
          className="fa fa-home"
          style={
            Object {
              "marginLeft": "10px",
            }
          }
        />
        <span
          style={
            Object {
              "fontSize": "13px",
            }
          }
        >
          view1
        </span>
        <span
          className="label label-primary pull-right"
        >
          1
        </span>
      </a>
    </li>
    <li
      className=""
    >
      <a
        href="/view2"
        onClick={[Function]}
      >
        <i
          className="fa fa-user"
          style={
            Object {
              "marginLeft": "10px",
            }
          }
        />
        <span
          style={
            Object {
              "fontSize": "13px",
            }
          }
        >
          view2
        </span>
      </a>
    </li>
  </ul>
</div>
`;
